packages/intrepid2/src/Projection/Intrepid2_LagrangianInterpolationDef.hpp

```diff
@@ -87,39 +90,39 @@ LagrangianInterpolation<SpT>::getDofCoordsAndCoeffs(
   Teuchos::RCP<Basis<SpT,scalarType,scalarType> > edgeBasis;
 
 
-  if (name == "Intrepid2_HGRAD_QUAD_Cn_FEM") {
+  if ((name == "Intrepid2_HGRAD_QUAD_Cn_FEM") || (name == "Intrepid2_HGRAD_QUAD_C1_FEM")) {
     edgeBasis = Teuchos::rcp(new Basis_HGRAD_LINE_Cn_FEM<SpT,scalarType,scalarType>(degree, pointType) );
-  } else if (name == "Intrepid2_HGRAD_HEX_Cn_FEM") {
+  } else if ((name == "Intrepid2_HGRAD_HEX_Cn_FEM") || (name == "Intrepid2_HGRAD_HEX_C1_FEM")){
     edgeBasis = Teuchos::rcp(new Basis_HGRAD_LINE_Cn_FEM<SpT,scalarType,scalarType>(degree, pointType) );
```

Comment:
Why not use Basis_HGRAD_LINE_C1_FEM if name=="Intrepid2_HGRAD_HEX_C1_FEM"? Same for other first order bases.

Reply:
It is possible, but I decided to simplify the code and not create more cases... This is only run once and it is independent on the batch side, so performance should not be an issue. Also we are considering refactorng all the basis, so that we can extract the sides/edges bases from the basis, so that we can get rid of all these conditionals

Reply:
Ok. That makes sense. But given your other comment that "HGRAD_HEX_C1 is different than HGRAD_HEX_Cn with degree=1", is it really ok?

Reply:
They are the same for LINE elements.

Reply:
Oh, I see. Ok.